pyquil/experiment/_memory.py

```diff
@@ -0,0 +1,259 @@
+    """
+    return pauli_term_to_euler_memory_map(term,
+                                          prefix=label,
+                                          tuple_x=P_X,
+                                          tuple_y=P_Y,
+                                          tuple_z=P_Z)
+
+
+def pauli_term_to_measurement_memory_map(
+        term: PauliTerm,
+        label: str = 'measurement',
+) -> Dict[str, List[float]]:
+    """
+    Given a ``PauliTerm``, create a memory map corresponding to the ZXZXZ-decomposed single-qubit
+    gates that allow for measurement in the eigenbasis of the ``PauliTerm``. For example, if we
+    have the following program:
+
+        RZ(measurement_alpha[0]) 0
+        RZ(pi/2) 0
+        RZ(measurement_beta[0]) 0
+        RX(-pi/2) 0
+        RZ(measurement_gamma[0]) 0
+        MEASURE 0 ro[0]
+
+    We can measure in the ``Y`` basis (by default we measure in the ``Z`` basis) by providing the
+    following memory map (which corresponds to ``RX(pi/2)``):
+
+        {'measurement_alpha': [pi/2], 'measurement_beta': [pi/2], 'measurement_gamma': [pi/2]}
+
+    :param term: The ``PauliTerm`` in question.
+    :param label: The prefix to provide to ``pauli_term_to_euler_memory_map``, for labeling the
+        declared memory regions. Defaults to "measurement".
+    :return: Memory map for measuring in the desired basis.
+    """
+    return pauli_term_to_euler_memory_map(term,
+                                          prefix=label,
+                                          tuple_x=M_X,
+                                          tuple_y=M_Y,
+                                          tuple_z=M_Z)
+
+
+def build_symmetrization_memory_maps(
+        memory_size: int,
+        symmetrization_level: int = -1,
+        label: str = 'symmetrization'
+) -> List[Dict[str, List[float]]]:
+    """
+    Build a list of memory maps to be used in a program that is trying to perform readout
+    symmetrization via parametric compilation. For example, if we have the following program:
+
+        RX(symmetrization[0]) 0
+        RX(symmetrization[1]) 1
+        MEASURE 0 ro[0]
+        MEASURE 1 ro[1]
+
+    We can perform exhaustive readout symmetrization on our two qubits by providing the four
+    following memory maps, and then appropriately flipping the resultant bitstrings:
+
+        {'symmetrization': [0.0, 0.0]} -> XOR results with [0,0]
+        {'symmetrization': [0.0, pi]}  -> XOR results with [0,1]
+        {'symmetrization': [pi, 0.0]}  -> XOR results with [1,0]
+        {'symmetrization': [pi, pi]}   -> XOR results with [1,1]
+
+    :param memory_size: Size of the memory region to symmetrize.
+    :param symmetrization_level: Level of symmetrization to perform. See ``SymmetrizationLevel``.
+    :param label: Name of the declared memory region. Defaults to "symmetrization".
+    :return: List of memory maps that performs the desired level of symmetrization.
+    """
+    if symmetrization_level == 0:
```

Comment:
Same here and below, let's use constants in `SymmetrizationLevel` for the sake of readability.

Reply:
same issue above, ill probably try and move things around